zboxcore/sdk/allocation.go

```diff
@@ -413,6 +433,66 @@ func (a *Allocation) uploadOrUpdateFile(localpath string, remotepath string,
 	return nil
 }
 
+func (a *Allocation) uploadOrUpdateUsingStream(input *UploadStreamInput, remotepath string,
+	status StatusCallback, isUpdate bool, thumbnailpath string, encryption bool,
+	isRepair bool, attrs fileref.Attributes) error {
+
+	if !a.isInitialized() {
+		return notInitialized
+	}
+
+	remotepath = zboxutil.RemoteClean(remotepath)
+	isabs := zboxutil.IsRemoteAbs(remotepath)
+	if !isabs {
+		return common.NewError("invalid_path", "Path should be valid and absolute")
+	}
+	remotepath = zboxutil.GetFullRemotePathForStreamUpload(remotepath)
```

Comment:
Is this different from zboxutil.GetFullRemotePath(remotepath, remotepath)?

Reply:
Yes, GetFullRemotePath(remoteParth, remotePath) would produce the same result, but it would not be a correct representation of the function. I can change it, but my opinion is that it'd create confusion for anyone who works on it at a later date.

Reply:
You have a point, but block copied code is bad and should be avoided. Can we rewrite `GetFullRemotePathForStreamUpload` so the code is not duplicated. For example
```go
func GetFullRemotePathForStreamUpload(remotePath string) string {
   GetFullRemotePath(remotePath, remotePath)
}
```

Reply:
yes, this is a better solution.